internal/runner/nomad_manager.go

```diff
@@ -95,69 +100,158 @@ func (m *NomadRunnerManager) SynchronizeRunners(ctx context.Context) error {
 	return err
 }
 
+func (m *NomadRunnerManager) StoreEnvironment(environment ExecutionEnvironment) {
+	m.AbstractManager.StoreEnvironment(environment)
+	m.reloadingEnvironment.Add(environment.ID().ToString(), &alertData{Mutex: &sync.Mutex{}, cancel: nil})
+}
+
+func (m *NomadRunnerManager) DeleteEnvironment(id dto.EnvironmentID) {
+	m.AbstractManager.DeleteEnvironment(id)
+	m.reloadingEnvironment.Delete(id.ToString())
+}
+
+// checkPrewarmingPoolAlert checks if the prewarming pool contains enough idle runners as specified by the PrewarmingPoolThreshold
+// if not it starts an environment reload mechanism according to the PrewarmingPoolReloadTimeout.
+func (m *NomadRunnerManager) checkPrewarmingPoolAlert(environment ExecutionEnvironment, runnerAdded bool) {
+	data, ok := m.reloadingEnvironment.Get(environment.ID().ToString())
+	if !ok {
+		log.WithField(dto.KeyEnvironmentID, environment.ID()).Error("reloadingEnvironment not initialized")
+		return
+	}
+
+	if runnerAdded && data.cancel != nil {
+		data.cancel()
+		data.cancel = nil
+		m.checkPrewarmingPoolAlert(environment, false)
+		return
+	}
+
+	// With this hard lock we collect/block goroutines waiting for one reload to be done.
+	// However, in practice its likely that only up to PrewarmingPoolSize/2 goroutines are waiting.
+	// We could avoid the waiting, but we use it to solve the race conditions of the recursive call above.
+	data.Lock()
+	defer data.Unlock()
+
+	prewarmingPoolThreshold := config.Config.Server.Alert.PrewarmingPoolThreshold
+	reloadTimeout := config.Config.Server.Alert.PrewarmingPoolReloadTimeout
+
+	if reloadTimeout == 0 || float64(environment.IdleRunnerCount())/float64(environment.PrewarmingPoolSize()) >= prewarmingPoolThreshold {
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	data.cancel = cancel
+
+	log.WithField(dto.KeyEnvironmentID, environment.ID()).Info("Prewarming Pool Alert. Checking again..")
```

Comment:
So far, we mostly used three dots:

```suggestion
	log.WithField(dto.KeyEnvironmentID, environment.ID()).Info("Prewarming Pool Alert. Checking again...")
```

When checking the code, we sometimes have a space before the three dots, sometimes not. Should we unify that (just in a dedicated commit, within this PR)?